central/delegatedregistryconfig/delegator/delegator.go

```diff
@@ -0,0 +1,148 @@
+package delegator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/stackrox/rox/central/delegatedregistryconfig/datastore"
+	"github.com/stackrox/rox/central/sensor/service/connection"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/urlfmt"
+	"github.com/stackrox/rox/pkg/waiter"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// New creates a new delegator.
+func New(deleRegConfigDS datastore.DataStore, connManager connection.Manager, scanWaiterManager waiter.Manager[*storage.Image]) *delegatorImpl {
+	return &delegatorImpl{
+		deleRegConfigDS:   deleRegConfigDS,
+		connManager:       connManager,
+		scanWaiterManager: scanWaiterManager,
+	}
+}
+
+type delegatorImpl struct {
+	// deleRegConfigDS for pulling the current delegated registry config.
+	deleRegConfigDS datastore.DataStore
+
+	// connManager for sending scan requests to secured clusters and ensuring
+	// clusters are valid for delegation.
+	connManager connection.Manager
+
+	// scanWaiterManager creates waiters that wait for async scan responses.
+	scanWaiterManager waiter.Manager[*storage.Image]
+}
+
+// GetDelegateClusterID returns the cluster id that should enrich this image (if any) and
+// true if enrichment should be delegated to a secured cluster, false otherwise.
+func (d *delegatorImpl) GetDelegateClusterID(ctx context.Context, image *storage.Image) (string, bool, error) {
+	config, err := d.getConfig(ctx)
```

Comment:
preferably, just handle the exists case here then we are ensured that the call to should delete has a non nil config